src/Dotnet.Script.Core/ScriptPublisher.cs

```diff
@@ -63,29 +63,32 @@ public ScriptPublisher(LogFactory logFactory, ScriptEmitter scriptEmitter)
             }
         }
 
-        public void CreateExecutable<TReturn, THost>(ScriptContext context, LogFactory logFactory, string runtimeIdentifier)
+        public void CreateExecutable<TReturn, THost>(ScriptContext context, LogFactory logFactory, string runtimeIdentifier, string executeFileName = null)
         {
             if (runtimeIdentifier == null)
             {
                 throw new ArgumentNullException(nameof(runtimeIdentifier));
             }
 
+            executeFileName = executeFileName ?? Path.GetFileNameWithoutExtension(context.FilePath);
             const string AssemblyName = "scriptAssembly";
 
             var tempProjectPath = ScriptProjectProvider.GetPathToProjectFile(Path.GetDirectoryName(context.FilePath), ScriptEnvironment.Default.TargetFramework);
+            var renamedProjectPath = ScriptProjectProvider.GetPathToProjectFile(Path.GetDirectoryName(context.FilePath), ScriptEnvironment.Default.TargetFramework, executeFileName);
             var tempProjectDirectory = Path.GetDirectoryName(tempProjectPath);
-
+            
             var scriptAssemblyPath = CreateScriptAssembly<TReturn, THost>(context, tempProjectDirectory, AssemblyName);
+           
             var projectFile = new ProjectFile(File.ReadAllText(tempProjectPath));
             projectFile.PackageReferences.Add(new PackageReference("Microsoft.CodeAnalysis.Scripting", ScriptingVersion));
             projectFile.AssemblyReferences.Add(new AssemblyReference(scriptAssemblyPath));
-            projectFile.Save(tempProjectPath);
+            projectFile.Save(renamedProjectPath);
```

Comment:
i don't think we should *always* create two different paths. only when the executable file name was overridden, otherwise it should use the old logic to avoid the second unnecessary file?

Reply:
From @seesharper, we should default to the name of the script filename rather than `script`, so the overridden happens when there's no `-n` option passed. Only when the script name is `script.csx`,  the two path will be the same.